src/include/compat.h

```diff
@@ -110,4 +112,25 @@
 #define MSG_DONTWAIT MSG_NONBLOCK
 #endif
 
+#if defined(HAVE_PTHREAD_SETNAME_NP)
+#define PTHREAD_SETNAME_NP(thread, name) pthread_setname_np(thread, name)
```

Comment:
much better. imo, can we avoid having the all upper case function name, i.e. reuse the name `pthread_setname_np()`, and define it with its equivalent if it's not available? i know, people might want to argue that macro should be named as `FUNCTION()` instead of `function()`.